lib/puppet/provider/tpm2_ownership/tpm2tools.rb

```diff
@@ -0,0 +1,201 @@
+Puppet::Type.type(:tpm2_ownership).provide(:tpm2tools) do
+  desc 'tpm2tools providers uses the TCG software stack
+    and commands provided by tpm2-tools rpm to set the
+    authentication tokens for the tpm if it is a version 2
+    tpm.
+
+    @author SIMP Team https://simp-project.com'
+
+  has_feature :take_ownership
+
+  mk_resource_methods
+  defaultfor :kernel => :Linux
+
+  commands :tpm2_takeownership => 'tpm2_takeownership'
+  commands :tpm2_getcap        => 'tpm2_getcap'
+
+  def initialize(value={})
+    super(value)
+    @property_flush = {}
+  end
+
+
+  # Generate standard args for connecting to the TPM.  These arguements
+  # are common for most TPM2 commands.
+  #
+  # @return [String] Return a string of the tcti and the
+  # hex option if it is set.
+  def get_extra_args()
+    options = []
+
+    debug('tpm2_takeownership setting tcti args.')
+    case resource[:tcti]
```

Comment:
Using non-abrmd TCTI options is a (very) advanced edge case.  Even if the provider techinically permits this:  **_Everything_ in the tpm2 module's critical path _must_ use the [`tpm2-abrmd`][ta] service,** which has its own TCTI configuration.

**tpm2-abrmd** is needed to provide the [TSS TAB/RM layers][tabrm] that support multi-process, multi-user systems.  In the tpm2-tools stack, it's also useful as a single place to configure TCTI for the entire system.

A resource using a non-`abrmd` TCTI would need to: 
* handle its own context management (swapping data in and out of the limited TPM memory as needed, like a virtual memory manager)
* handle cases when competing processes are blocking direct access to the TPM

In the case of the TPM2 module:
* The `:tpm2` fact runs before the agent requests a Puppet catalog, so it relies on the tabrmd service's TCTI connection to communicate with the correct TPM.    
* As a consequence, the fact's data won't be useful to any `tpm2_ownership` resource with a TCTI that is different from the tabrmd service.


[ta]: https://github.com/tpm2-software/tpm2-abrmd/tree/1.x
[tabrm]: https://trustedcomputinggroup.org/wp-content/uploads/TSS-TAB-and-Resource-Manager-ver1.0-rev16_Public_Review.pdf